src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java

```diff
@@ -364,77 +365,291 @@ private void processBikeRentalNodes() {
+                    creativeName,
+                    entrances
+            );
+
+            VehicleParkingService vehicleParkingService = graph.getService(VehicleParkingService.class, true);
+            vehicleParkingService.addVehicleParking(vehicleParking);
+
+            VehicleParkingHelper.linkVehicleParkingToGraph(graph, vehicleParking);
+
+            return true;
+        }
+
+        private VehicleParking createVehicleParkingObjectFromOsmEntity(
+                boolean isCarParkAndRide,
+                double lon,
+                double lat,
+                OSMWithTags entity,
+                I18NString creativeName,
+                List<VehicleParking.VehicleParkingEntranceCreator> entrances
+        ) {
+            OptionalInt bicycleCapacity, carCapacity, wheelchairAccessibleCapacity;
+            if (isCarParkAndRide) {
+                carCapacity = parseCapacity(entity);
+                bicycleCapacity = parseCapacity(entity, "capacity:bike");
+                wheelchairAccessibleCapacity = parseCapacity(entity, "capacity:disabled");
+            } else {
+                bicycleCapacity = parseCapacity(entity);
+                carCapacity = OptionalInt.empty();
+                wheelchairAccessibleCapacity = OptionalInt.empty();
+            }
+
+            VehicleParking.VehiclePlaces vehiclePlaces = null;
+            if (bicycleCapacity.isPresent() || carCapacity.isPresent() || wheelchairAccessibleCapacity.isPresent()) {
+                vehiclePlaces = VehicleParking.VehiclePlaces.builder()
+                    .bicycleSpaces(bicycleCapacity.isPresent() ? bicycleCapacity.getAsInt() : null)
+                    .carSpaces(carCapacity.isPresent() ? carCapacity.getAsInt() : null)
+                    .wheelchairAccessibleCarSpaces(wheelchairAccessibleCapacity.isPresent() ? wheelchairAccessibleCapacity.getAsInt() : null)
+                    .build();
+            }
+
+            var bicyclePlaces = !isCarParkAndRide || bicycleCapacity.orElse(0) > 0;
+            var carPlaces = (
+                    isCarParkAndRide &&
+                            wheelchairAccessibleCapacity.isEmpty() && carCapacity.isEmpty()
+            ) || carCapacity.orElse(0) > 0;
+            var wheelchairAccessibleCarPlaces = wheelchairAccessibleCapacity.orElse(0) > 0;
+
+            var id = new FeedScopedId(
+                    VEHICLE_PARKING_OSM_FEED_ID,
+                    String.format("%s/%d", entity.getClass().getSimpleName(), entity.getId())
+            );
+
+            return VehicleParking.builder()
+                    .id(id)
+                    .name(creativeName)
+                    .x(lon)
+                    .y(lat)
+                    .bicyclePlaces(bicyclePlaces)
+                    .carPlaces(carPlaces)
+                    .wheelchairAccessibleCarPlaces(wheelchairAccessibleCarPlaces)
+                    .capacity(vehiclePlaces)
+                    .entrances(entrances)
+                    .build();
```

Comment:
I would find it useful to add a tag that the parking lot is sourced from OSM. WDYT?

Reply:
Actually, it's already in the id, so probably not needed.

Reply:
My thought was that it's present in the `id`, and so is not needed explicitly.